src/Model/PatientDictContainer.py

```diff
@@ -23,6 +23,12 @@ def __init__(self, path, dataset, filepaths, **kwargs):
         self.filepaths = filepaths
         self.additional_data = kwargs
 
+    def add(self, key, value):
```

Comment:
implies silent overwrite.
If the PatientDictContainer is expected to be sort of immutable, the "additional_data" part is not going to act that way now that there is an add that overwrites.
Depending on the behaviour you want, it might be prudent to raise a KeyError if someone tries to add a key that is already there.